src/system.py

```diff
@@ -315,18 +315,249 @@ def __init__(self):
+                                'RFTS': '>u1',
+                                'FRAM': '>u1',
+                                'SEC': '>u1',
+                                'CTS': '>u1',
+                                'Boot': '>u1',
+                                'Pipe': '>u1',
+                                'BCN': '>u1',
+                                'Echo': '>u1',
+                                'RF Mode1': '>u1',  # Use table 9 to improve
+                                'RF Mode2': '>u1',
+                                'RF Mode3': '>u1',
+                                'Reset': '>u1',
+                                'Frequency': '>u4',
+                                'PIPE timeout': '>u2',
+                                'Beacon period': '>u2',
+                                'Audio trans. period': '>u2',
+                                'Uptime': '>u4',
+                                'Packets out': '>u4',
+                                'Packets in': '>u4',
+                                'Packets in CRC16': '>u4',
+                                'Temperature': '>f',
+                                'Low power status': '>u1',
+                                'Firmware Version': '>u1',
+                                'Payload Size': '>u2',
+                                'Secure key': '>u4',
+                            }
+                        }
+                    },
+                    'UHF_GET_CALLSIGN': {
+                        'subPort': 37,
+                        'inoutInfo': {
+                            'args': None,
+                            'returns': {
+                                'err': '>b',
+                                'Destination': '>U6',
+                                'Source': '>U6',
+                            }
+                        }
+                    },
+                    'UHF_GET_MORSE': {
+                        'subPort': 38,
+                        'inoutInfo': {
+                            'args': None,
+                            'returns': {
+                                'err': '>b',
+                                'Morse': '>U36',
+                            }
+                        }
+                    },
+                    'UHF_GET_MIDI': {
+                        'subPort': 39,
+                        'inoutInfo': {
+                            'args': None,
+                            'returns': {
+                                'err': '>b',
+                                'MIDI': '>U36',
+                            }
+                        }
+                    },
+                    'UHF_GET_BEACON_MSG': {
+                        'subPort': 40,
+                        'inoutInfo': {
+                            'args': None,
+                            'returns': {
+                                'err': '>b',
+                                'Beacon Message': '>U36',  # 120 too large
```

Comment:
For all of the beacon and call sign stuff, the EH allocates a length of 120. However, based on the manual, other than this one, the rest will be of length 6 or 36. However, when you allocate 120 for this one, it'll go beyond the buffer size and things do not work out. The best solution would be to somehow read the length of the message (similar to sending it in `UHF_SET_BEACON_MSG` automatically and set the length using that which I don't know how to do.

Also, the command parser doesn't allow spaces in the middle of the message which is not good. Any suggestions for that?

Reply:
In the middle of the message, as in a string? Should be able to add a space into the regular expression